src/util/strconv.h

```diff
@@ -106,25 +110,98 @@ static char *rm_strndup_unescape(const char *s, size_t len) {
   return ret;
 }
 
-// strndup + lowercase in one pass!
-static char *rm_strdupcase(const char *s, size_t len) {
+// transform utf8 string to lower case using nunicode library
+// encoded: the utf8 string to transform, if the transformation is successful
+//          the transformed string will be written back to this buffer
+// in_len: the length of the utf8 string
+// returns the bytes written to encoded, or 0 if the length of the transformed
+// string is greater than in_len and no transformation was done
+static size_t unicode_tolower(char *encoded, size_t in_len) {
+  uint32_t u_stack_buffer[SSO_MAX_LENGTH];
+  uint32_t *u_buffer = u_stack_buffer;
+
+  if (in_len == 0) {
+    return 0;
+  }
+
+  const char *encoded_char = encoded;
+  ssize_t u_len = nu_strtransformnlen(encoded, in_len, nu_utf8_read,
+                                              nu_tolower, nu_casemap_read);
+
+  if (u_len >= (SSO_MAX_LENGTH - 1)) {
+    u_buffer = (uint32_t *)rm_malloc(sizeof(*u_buffer) * (u_len + 1));
+  }
+
+  // Decode utf8 string into Unicode codepoints and transform to fold
```

Comment:
Doing tolower now instead of fold
```suggestion
  // Decode utf8 string into Unicode codepoints and transform to lower
```